### PR TITLE
Handle more socket exceptions

### PR DESCRIPTION
**Pull request by dabcoder**
Attempts to fix https://github.com/DataDog/datadogpy/issues/212.

### CODE REVIEW

File: datadog/dogstatsd/base.py [hunk 1]
```diff
@@ -289,9 +289,12 @@ def _send_to_server(self, packet):
         except socket.timeout:
             # dogstatsd is overflowing, drop the packets (mimicks the UDP behaviour)
             return
-        except socket.error:
+        except (socket.error, socket.herror, socket.gaierror) as se:
             log.info("Error submitting packet, dropping the packet and closing the socket")
```
Comment (gzussa) [inline datadog/dogstatsd/base.py]: What about changing this to a `warning` instead?

File: datadog/dogstatsd/base.py [hunk 2]
```diff
@@ -289,9 +289,12 @@ def _send_to_server(self, packet):
...
             self.close_socket()
+        except Exception as e:
+            log.info("Unexpected error: ", e)
```
Comment (gzussa) [inline datadog/dogstatsd/base.py]: Actually maybe sending `error` would be more appropriate.

File: datadog/dogstatsd/base.py [hunk 3]
```diff
@@ -289,9 +289,12 @@ def _send_to_server(self, packet):
         except socket.timeout:
             # dogstatsd is overflowing, drop the packets (mimicks the UDP behaviour)
             return
-        except socket.error:
+        except (socket.error, socket.herror, socket.gaierror) as se:
             log.info("Error submitting packet, dropping the packet and closing the socket")
             self.close_socket()
```
Comment (nmuesch) [inline datadog/dogstatsd/base.py]: Should we be closing the socket in the second `except` clause as well?

Comment (dabcoder) [reply datadog/dogstatsd/base.py]: Not sure, `# If set, use socket directly`, would the last exception cover the case of not using the socket and thus not needing to close it? Or am I on the wrong track there?

File: datadog/dogstatsd/base.py [hunk 4]
```diff
@@ -289,9 +293,12 @@ def _send_to_server(self, packet):
         except socket.timeout:
             # dogstatsd is overflowing, drop the packets (mimicks the UDP behaviour)
             return
-        except socket.error:
-            log.info("Error submitting packet, dropping the packet and closing the socket")
+        except (socket.error, socket.herror, socket.gaierror) as se:
+            log.warning("Error submitting packet: {}, dropping the packet and closing the socket".format(str(se)))
```
Comment (ofek) [inline datadog/dogstatsd/base.py]:
```suggestion
            log.warning("Error submitting packet: {}, dropping the packet and closing the socket".format(se))
```

**Pull request merged**